### PR TITLE
add --without-comments option for show

### PR DESCRIPTION
**Pull request by reidliu41**
- add --without-comments option and it will remove the comments from the show output, and make it easy to view

```
$ ilab config show --help
Usage: ilab config show [OPTIONS]

  Displays the current config as YAML with an option to extract a specific
  key.

Options:
  -k, --key TEXT          Show only a specific section of the configuration,
                          e.g., -k chat or -k chat.context
  -wc, --without-comments  Show the config without comments. Can be used in
                           conjunction with -k.
  --help                  Show this message and exit.


# ilab config show
# Chat configuration section.
chat:
  # Predefined setting or environment that influences the behavior and responses of
  # the chat assistant. Each context is associated with a specific prompt that
  # guides the assistant on how to respond to user inputs.
  # Default: default
  context: default
  # Sets temperature to 0 if enabled, leading to more deterministic responses.
  # Default: False
  greedy_mode: false
  # Directory where chat logs are stored.
  # Default: /root/.local/share/instructlab/chatlogs
  logs_dir: /root/.local/share/instructlab/chatlogs
  # The maximum number of tokens that can be generated in the chat completion. Be
  # aware that larger values use more memory.
  # Default: None
  max_tokens:
....

# ilab config show -wc
chat:
  context: default
  greedy_mode: false
  logs_dir: /root/.local/share/instructlab/chatlogs
  max_tokens:
  model: /root/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf
  session:
  vi_mode: false
  visible_overflow: true
evaluate:
  base_branch:
  base_model: instructlab/granite-7b-lab
  branch:
  gpus:
  mmlu:
    batch_size: auto
    few_shots: 5
  mmlu_branch:
    tasks_dir: /root/.local/share/instructlab/datasets
  model: /root/.cache/instructlab/models/merlinite-7b-lab-Q4_K_M.gguf
  mt_bench:
    judge_model: prometheus-eval/prometheus-8x7b-v2.0
    max_workers: auto
    output_dir: /root/.local/share/instructlab/internal/eval_data
  mt_bench_branch:
    taxonomy_path: /root/.local/share/instructlab/taxonomy
....


# ilab config show -k general
  # Debug level for logging.
  # Default: 0
debug_level: 0
  # Log format. https://docs.python.org/3/library/logging.html#logrecord-attributes
  # Default: %(levelname)s %(asctime)s %(name)s:%(lineno)d: %(message)s
log_format: '%(levelname)s %(asctime)s %(name)s:%(lineno)d: %(message)s'
  # Log level for logging.
  # Default: INFO
log_level: INFO

# ilab config show -k general -wc
debug_level: 0
log_format: '%(levelname)s %(asctime)s %(name)s:%(lineno)d: %(message)s'
log_level: INFO
```

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by nathan-weinberg — APPROVED** [review]

File: src/instructlab/config/show.py [hunk 1]
```diff
@@ -33,8 +43,16 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
     default=None,
     help="Show only a specific section of the configuration, e.g., -k chat or -k chat.context",
 )
+@click.option(
+    "--remove-comments",
```
Comment (danmcp) [inline src/instructlab/config/show.py:47]: Would --without-comments be better?

Then is would read as

`ilab config show --without-comments`

vs

`ilab config show --remove-comments`

File: tests/test_lab_config.py [hunk 2]
```diff
@@ -51,6 +57,34 @@ def test_ilab_config_show_key_general_yaml(cli_runner: CliRunner) -> None:
     assert actual_output == expected_yaml_output
 
 
+def test_ilab_config_show_key_remove_comments(cli_runner: CliRunner) -> None:
+    cfg = configuration.get_default_config()
+    key_config_commented_map = configuration.config_to_commented_map(cfg.general)
+
+    yaml_stream = ruamel.yaml.compat.StringIO()
```
Comment (danmcp) [inline tests/test_lab_config.py:64]: Logic could be shared with https://github.com/instructlab/instructlab/pull/2602/files in a common method

File: src/instructlab/config/show.py [hunk 3]
```diff
@@ -24,6 +25,15 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
     return data_dict
 
 
+def show_config_without_comments(output):
+    """Recursively show config without comments from a ruamel.yaml data structure."""
+    if isinstance(output, CommentedMap):
+        output.ca.items.clear()
+        for key in output:
+            if isinstance(output[key], (CommentedMap, list)):
```
Comment (danmcp) [inline src/instructlab/config/show.py:33]: Why include list here if show_config_without_comments doesn't do anything with lists?

File: src/instructlab/config/show.py [hunk 4]
```diff
@@ -43,6 +61,9 @@ def show(ctx: click.Context, key_path: str) -> None:
             commented_map if not key_path else get_nested_value(commented_map, key_path)
         )
 
+        if without_comments:
+            show_config_without_comments(result)
```
Comment (danmcp) [inline src/instructlab/config/show.py:65]: Suggestion for a better name: `strip_comments` ?

File: src/instructlab/config/show.py [hunk 5]
```diff
@@ -24,6 +25,15 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
     return data_dict
 
 
+def show_config_without_comments(output):
```
Comment (danmcp) [inline src/instructlab/config/show.py:28]: Would be good to add typing info to this function

Comment (danmcp) [inline src/instructlab/config/show.py:28]: Variable might be better named as config?

File: src/instructlab/config/show.py [hunk 6]
```diff
@@ -24,6 +25,15 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
...
+    """Recursively show config without comments from a ruamel.yaml data structure."""
```
Comment (danmcp) [inline src/instructlab/config/show.py:29]: If the method is renamed, should adjust this comment as well.

File: src/instructlab/config/show.py [hunk 7]
```diff
@@ -24,6 +25,15 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
...
+    if isinstance(output, CommentedMap):
+        output.ca.items.clear()
+        for key in output:
```
Comment (danmcp) [inline src/instructlab/config/show.py:32]: Could avoid the key lookups with:

```
        for value in output.values():
            if isinstance(value, CommentedMap):
                show_config_without_comments(value)
```

**Review by danmcp — CHANGES_REQUESTED** [review]

File: src/instructlab/config/show.py [hunk 8]
```diff
@@ -24,6 +25,15 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
     return data_dict
 
 
+def strip_comments(config: CommentedMap) -> None:
+    """Recursively remove comments from a ruamel.yaml CommentedMap structure."""
+    if isinstance(config, CommentedMap):
```
Comment (danmcp) [inline src/instructlab/config/show.py:30]: Is this check necessary?  If it is, it seems like the typing for the function is wrong.

File: src/instructlab/config/show.py [hunk 9]
```diff
@@ -24,6 +25,14 @@ def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
     return data_dict
 
 
+def strip_comments(config: CommentedMap) -> None:
```
[the remark this answers is not in the document]
Comment (reidliu41) [reply src/instructlab/config/show.py:28]: seems not work for the last level, add the check is fine
```
 ilab config show -wc -k serve.vllm
gpus:
llm_family: ''
max_startup_attempts: 120

$ ilab config show -wc -k serve.vllm.gpus
Traceback (most recent call last):
AttributeError: 'NoneType' object has no attribute 'ca'
```

[the remark this answers is not in the document]
Comment (danmcp) [reply src/instructlab/config/show.py:28]: Sorry not following.  Are you saying it should fail that way for serve.vllm.gpus?

[the remark this answers is not in the document]
Comment (reidliu41) [reply src/instructlab/config/show.py:28]: CommentedMap | Any  + isinstance would be better

[the remark this answers is not in the document]
Comment (danmcp) [reply src/instructlab/config/show.py:28]: I can live with that.  Do you not like the option of calling isinstance(result, CommentedMap) before calling strip_comments from show() ?

[the remark this answers is not in the document]
Comment (reidliu41) [reply src/instructlab/config/show.py:28]: no... I think is ok too..haha
I just think you need to type "Any" , actually i used isinstance(result, CommentedMap) before. ok, let check it before the calling.

**Review by danmcp — APPROVED** [review]

**Pull request merged**